src/ocrmypdf/_graft.py

```diff
@@ -307,7 +311,11 @@ def _graft_text_layer(
 
             base_resources = _ensure_dictionary(base_page.obj, Name.Resources)
             base_xobjs = _ensure_dictionary(base_resources, Name.XObject)
-            text_xobj_name = Name.random(prefix="OCR-")
+            if self.context.options.deterministic_output:
+                # Use a stable name per page for deterministic output
+                text_xobj_name = Name(f"/OCR-{page_num:06d}")
```

Comment:
I should bind QPDFObjectHandle::getUniqueResourceName in pikepdf to make this easier.

We can't count on any particular name being available (in particular, if OCRmyPDF is being reused on the same file in some weird way, or if the input file is merged from multiple files generated by OCRmyPDF).

Without have getUniqueResourceName, just make a prefix, appending a counter, and keep incrementing until there's no conflict with an existing name.